File: anylang/c++/2236.判断根结点是否等于子结点之和.cpp
#include <stdbool.h>
#include <stdio.h>

#include "tree.hpp"

class Solution {
   public:
    bool checkTree(TreeNode* root) {
        return root->val == root->right->val + root->left->val ? true : false;
    }
};

int main() {
    struct TreeNode root, left, right;
    root.val = 3;
    left.val = 1;
    right.val = 2;
    root.left = &left;
    root.right = &right;

    struct TreeNode root2 = {
        3,
        new TreeNode(3),
        new TreeNode(1)};

    Solution s;

    bool rootRst = s.checkTree(&root);
    bool root2Rst = s.checkTree(&root2);

    printf("rootRst is : %s\n", rootRst ? "true" : "false");
    printf("root2Rst is : %s\n", root2Rst ? "true" : "false");

    return 0;
}

File: anylang/go/leetcode/2236.判断根结点是否等于子结点之和.go
package leetcode

import "fmt"

func checkTree(root *TreeNode) bool {
	return root.Val == root.Left.Val+root.Right.Val
}

func CheckTree() {

	root := TreeNode{
		Val: 3,
		Left: &TreeNode{
			Val: 2,
		},
		Right: &TreeNode{
			Val: 1,
		},
	}

	fmt.Println("CheckTree: ", checkTree(&root))
}
